app/services/record_declarations/base.rb

```diff
@@ -26,6 +26,15 @@ def profile_exists
       end
     end
 
+    def date_has_the_right_format
+      return if declaration_date.blank?
+
+      errors.add(:declaration_date, I18n.t(:invalid_declaration_date)) unless declaration_date.match(RCF3339_DATE_REGEX)
```

Comment:
if `invalid_declaration_date` is a new error message i expected a new respective entry in the locales file